[FIX] base: compute object_write equation for each record

nle-odoo opened the pull request:
Scenario:
- create an object_write (Update the Record) action
- update a field with equation evaluation (Python expression) with an
  expression like "record.id"
- click on "Create contextual action"
- go to a list view of the model, execute the action for several records

Result: the value of the field of all records are computed based on the
first selected record.

Issue: we don't change the "record" in the evaluation context and just
keep the first record.

Fix: changing the record.

Note: without the fix, the added test fails because the city of the
second record is set to the value of the ID of the first record.

opw-4491099

__PR code note__:

I did the change in "run" but this could be done in `_run_action_object_write`, I chose run because we are already changing the action_id in it.

The "`if eval_context.get('record') is not None`" is to not change an hypothetical case where `self.model_id` doesn't match the `context.action_model`.

__PR note__:

If it was too risky, we could probably merge in an higher version, the ticket is in 17.0 and this is still happening in master.

I've not heard of this issue before so this must not be very frequent or urgent, but I guess this is because people either use a python constant or they just use a python action with a for loop in general.

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/210855.png)](https://mergebot.odoo.com/odoo/odoo/pull/210855)

nle-odoo commented:
I updated to also update `records` and I removed the "if" since it was very unlikely that it would be useful (it was a very hypothetical case of having a different active_model in context than the action), and anyway for example the `_run_action_object_write` will write on the browsed active_id so it makes sense to browse on it for the record in my opinion:

https://github.com/odoo/odoo/blob/c4321027052699caa359a602f5a449689ab6fafb/odoo/addons/base/models/ir_actions.py#L564

nle-odoo commented:
@odoo/rd-security I think the security check can be ignored: we already had in the eval context `{'record': model.browse(active_id), 'records': model.browse(active_ids)}`, this is just changing it to browse one id at a time and not always the same one

nle-odoo commented:
robodoo r+
